### PR TITLE
Add docs to project

### PR DESCRIPTION
**Pull request by jmichelsen**
Closes issue #12

### CODE REVIEW

File: docs/best_practices.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+- Data Collection (**Yes**)
+    - *Views*/APIs (**get** and **post** methods and Django class-based views excluded)
+    - *Forms* (**customized validation**, any customizations to Django base forms)
+- 3rd Party Tools (**No**)
+    - Django ORM
+    - Python Library
+- Algorithm (**Yes**)
+- Data Manipulation (**Yes**)
+
+## Should This Be Mocked?
```
Comment (jmast02) [inline docs/best_practices.md]: What is the term 'Mocked' used for?

File: docs/best_practices.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+- Non-local Implementations (**Yes**)
+    - Communicating with 3rd party systems
+    
+## Package vs. Module
```
Comment (jmast02) [inline docs/best_practices.md]: Would definitely like to go over handlers/parsers to get a better understanding of those.

File: docs/best_practices.md (hunk 1 above)
Comment (jmichelsen) [reply docs/best_practices.md]: a Mock is a fake thing that acts like a real thing, used for testing. For example, when you want to test that your system can query an external API and manipulate the data it got back, but imagine you don't have an internet connection...you could mock the part of your application that actually does a request to the external system, and return data from a file instead.

There are mocking utilities built into python and Django to help this process. We'll get into this more when we do some unittests.

File: docs/best_practices.md (hunk 2 above)
Comment (jmichelsen) [reply docs/best_practices.md]: Parsers take data in one format and convert it to another format. It's a translation layer.

**Comment by jmichelsen** [comment]
closes #12

**Pull request merged**